### PR TITLE
doc: ARC: add release notes for ARC

### PR DESCRIPTION
**Pull request by kokas-a**
Add ARC release notes for 3.6 Zephyr release

### CODE REVIEW

**Comment by kokas-a** [comment]
@abrodkin , could you please take a look?

**Comment by MaureenHelm** [comment]
Why is this in draft state?

File: doc/releases/release-notes-3.6.rst [hunk 1]
```diff
@@ -29,6 +29,10 @@ Architectures
 
 * ARC
 
+ * Extended support for ARCv3 features: enable HW prefetch and shared cache;
+ * Disabled TLS support for platforms with two or more register banks;
```
Comment (abrodkin) [inline doc/releases/release-notes-3.6.rst:33]: Please explain what that TLS is. I guess here it stands for "Thread-local Storage". But there're other uses of the same acronym like "Transport Layer Security" etc. Let's not confuse a reader here.

File: doc/releases/release-notes-3.6.rst [hunk 2]
```diff
@@ -29,6 +29,10 @@ Architectures
 
 * ARC
 
+ * Extended support for ARCv3 features: enable HW prefetch and shared cache;
```
Comment (abrodkin) [inline doc/releases/release-notes-3.6.rst:32]: Improvements for ARCv3 processors (HS5x & HS6x): enable hardware prefetcher and shared cluster cache (SCM - Shared Cluster Memory).

File: doc/releases/release-notes-3.6.rst [hunk 3]
```diff
@@ -29,6 +29,10 @@ Architectures
...
+ * Disabled TLS support for platforms with two or more register banks;
+ * Fixed unstable work of images built with MWDT on HW platforms (garbage in .device_states section).
```
Comment (abrodkin) [inline doc/releases/release-notes-3.6.rst:34]: Fixed unstable work of application built with MetaWare toolchain for hardware platforms (garbage in .device_states section).

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by abrodkin — CHANGES_REQUESTED** [review]

**Comment by abrodkin** [comment]
> Why is this in draft state?

@MaureenHelm that's because we don't want it to be merged before another person (either @ruuddw or @evgeniy-paltsev) do their review. Last time our release notes were merged almost immediately before I had a chance to add my notes and we had to post a fix then, see https://github.com/zephyrproject-rtos/zephyr/pull/59321 ;)

**Comment by henrikbrixandersen** [comment]
> @MaureenHelm that's because we don't want it to be merged before another person (either @ruuddw or @evgeniy-paltsev) do their review. Last time our release notes were merged almost immediately before I had a chance to add my notes and we had to post a fix then, see #59321 ;)

Not submitting the release notes the day before the release is the best way to avoid that ;)

**Review by ruuddw — APPROVED** [review]

**Review by abrodkin — APPROVED** [review]

**Comment by abrodkin** [comment]
@MaureenHelm this one is ready for merge.

**Review by MaureenHelm — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**